src/logs/diagnostic-channel/initialization.ts

```diff
@@ -4,14 +4,17 @@
 // Don't reference modules from these directly. Use only for types.
 import * as DiagChannelPublishers from "diagnostic-channel-publishers";
 import { Logger } from "../../shared/logging";
+import { ShimJsonConfig } from "../../shim/jsonConfig";
```

Comment:
JSON config should never used outside of our Internal config class, value could be overridden using JS code so this ignores that

Reply:
noDiagnosticChannel and noPatchModules cannot be configured via `defaultClient.config.noDiagnosticChannel` or `defaultClient.config.noPatchModules`, only via env var or JSON config, so I don't believe this could be overridden before being read here. Were you referring to another way it could be overridden?